src/reactviews/pages/SchemaCompare/SchemaCompareStateProvider.tsx

```diff
@@ -0,0 +1,149 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as sc from "../../../sharedInterfaces/schemaCompare";
+import * as mssql from "vscode-mssql";
+
+import { createContext } from "react";
+import { useVscodeWebview } from "../../common/vscodeWebviewProvider";
+
+const schemaCompareContext = createContext<sc.SchemaCompareContextProps>(
+    {} as sc.SchemaCompareContextProps,
+);
+
+interface SchemaCompareStateProviderProps {
+    children: React.ReactNode;
+}
+
+const SchemaCompareStateProvider: React.FC<SchemaCompareStateProviderProps> = ({
+    children,
+}) => {
+    const webViewState = useVscodeWebview<
+        sc.SchemaCompareWebViewState,
+        sc.SchemaCompareReducers
+    >();
+    const schemaCompareState = webViewState?.state;
+
+    return (
+        <schemaCompareContext.Provider
+            value={{
+                state: schemaCompareState,
+                themeKind: webViewState?.themeKind,
+                schemaCompare: function (
+                    operationId: string,
+                    sourceEndpointInfo: mssql.SchemaCompareEndpointInfo,
+                    targetEndpointInfo: mssql.SchemaCompareEndpointInfo,
+                    taskExecutionMode: mssql.TaskExecutionMode,
+                    deploymentOptions: mssql.DeploymentOptions,
+                ): void {
+                    webViewState?.extensionRpc.action("schemaCompare", {
+                        operationId: operationId,
+                        sourceEndpointInfo: sourceEndpointInfo,
+                        targetEndpointInfo: targetEndpointInfo,
+                        taskExecutionMode: taskExecutionMode,
+                        deploymentOptions: deploymentOptions,
+                    });
+                },
+                schemaCompareGenerateScript: function (
+                    operationId: string,
+                    targetServerName: string,
+                    targetDatabaseName: string,
+                    taskExecutionMode: mssql.TaskExecutionMode,
+                ): void {
+                    webViewState?.extensionRpc.action(
+                        "schemaCompareGenerateScript",
+                        {
+                            operationId: operationId,
+                            targetServerName: targetServerName,
+                            targetDatabaseName: targetDatabaseName,
+                            taskExecutionMode: taskExecutionMode,
+                        },
+                    );
+                },
+                schemaComparePublishDatabaseChanges: function (
```

Comment:
since these are all part of the schemaCompare controller/provider, we probably don't need to prefix everything with "schemaCompare"